wear/src/main/java/io/homeassistant/companion/android/complications/ComplicationReceiver.kt

```diff
@@ -0,0 +1,134 @@
+package io.homeassistant.companion.android.complications
+
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import androidx.wear.watchface.complications.datasource.ComplicationDataSourceUpdateRequester
+import dagger.hilt.android.AndroidEntryPoint
+import io.homeassistant.companion.android.common.data.integration.Entity
+import io.homeassistant.companion.android.common.data.integration.IntegrationRepository
+import io.homeassistant.companion.android.database.AppDatabase
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class ComplicationReceiver : BroadcastReceiver() {
+    private var entityUpdates: Flow<Entity<*>>? = null
+
+    @Inject
+    lateinit var integrationUseCase: IntegrationRepository
+    private val scope = CoroutineScope(SupervisorJob() + Dispatchers.Main.immediate)
+    private var mainScope: CoroutineScope = CoroutineScope(Dispatchers.Main + Job())
+
+    override fun onReceive(context: Context, intent: Intent) {
+        val result = goAsync()
+
+        try {
+            when (intent.action.toString()) {
```

Comment:
the `.toString()` is not necessary here, because `intent.action` is already a `String`, and the potential `"null"` (`String`) return value this extension can produce, won't be used here anyways.